pilot/pkg/networking/plugin/plugin.go

```diff
@@ -44,17 +50,58 @@ const (
 	Health = "health"
 	// Mixer is the name of the mixer plugin passed through the command line
 	Mixer = "mixer"
+
+	SMTP    = 25
+	DNS     = 53
+	MySQL   = 3306
+	MongoDB = 27017
 )
 
-// ModelProtocolToListenerProtocol converts from a config.Instance to its corresponding plugin.ListenerProtocol
-func ModelProtocolToListenerProtocol(p protocol.Instance) ListenerProtocol {
+var (
+	// Ports be skipped for protocol sniffing. Applications bound to these ports will be broken if
+	// protocol sniffing is enabled.
+	wellKnownPorts = map[int]struct{}{
+		SMTP:    {}, // SMTP
+		DNS:     {}, // DNS. Default TCP if not specified.
+		MySQL:   {}, // MySQL
+		MongoDB: {}, // MongoDB
+	}
+)
+
+func IsProtocolSniffingPort(node *model.Proxy, port *model.Port) bool {
```

Comment:
this code will result in an a lot of unnecessary redundant parsing for every listener/port for every proxy. Whats the point of having a common model then?